### PR TITLE
WIP(Switch implementation in js)

### PR DESCRIPTION
**Pull request by iamneha**
#161

### CODE REVIEW

**Comment by imbstack** [comment]
This is looking good so far! I think the next step would be to add the yaml from [the issue](https://github.com/taskcluster/json-e/issues/161#issue-243104748) to `specification.yml` and then run the js tests with `yarn test`. Once the tests pass again, we can start work on the Python bit! Let me know if you have any questions, I'll be around in irc and here!

File: src/index.js [hunk 1]
```diff
@@ -195,6 +195,18 @@ operators.$merge = (template, context) => {
   return Object.assign({}, ...value);
 };
 
+operators.$switch = (template, context) => {
+  checkUndefinedProperties(template, ['\\switch']);
+
+  let value = render(template['$switch'], context);
```
Comment (jonasfj) [inline src/index.js]: This should be evaluate as expression in the expression language, not as json-e expression... This is what `$if` does with interpreter...

File: src/index.js [hunk 2]
```diff
@@ -195,6 +195,18 @@ operators.$merge = (template, context) => {
...
+
+  if (!isString(template['$switch'])) {
```
Comment (jonasfj) [inline src/index.js]: The value resulting from interpreting this expression...

**Comment by djmitche** [comment]
We went with `$match` with a somewhat different syntax..

**Pull request closed without merging**